app/models/classified_listing.rb

```diff
@@ -109,4 +114,12 @@ def validate_category
   def create_slug
     self.slug = title.to_s.downcase.parameterize.tr("_", "") + "-" + rand(100_000).to_s(26)
   end
+
+  def assign_classified_listing_category
+    category_name = CATEGORIES_AVAILABLE.dig(attributes["category"], :name)
+    category = ClassifiedListingCategory.find_by(name: category_name)
+    return unless category
```

Comment:
It's "safe" to return, as the validation will fail anyway.

Reply:
I still want to add an error not to leave a user completely confused just in case :D

Reply:
There will be an error from the validation due to the `belongs_to`: `:classified_listing_category=>["must exist"]`. What else would you like to add?

Reply:
This should be enough :+1: